tests/subsys/logging/log_syst/src/mock_backend.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "mock_backend.h"
+#include <ztest.h>
+#include <logging/log_core.h>
+#include <logging/log_backend_std.h>
+#include <stdlib.h>
+
+static uint32_t log_format_current = CONFIG_LOG_BACKEND_MOCK_OUTPUT_DEFAULT;
+union log_msg2_generic *test_msg;
+
+static uint8_t mock_output_buf[1];
+uint8_t test_output_buf[256];
+int pos_in_buf;
+
+/**
+ * @brief This function takes in the characters and copies
+ * them in the temporary buffer.
+ *
+ * @return No. of characters copied to the temporary buffer.
+ */
+static int char_out(uint8_t *data, size_t length, void *ctx)
+{
+	ARG_UNUSED(ctx);
+	size_t output_length = length;
+
+	while (length > 0) {
+		test_output_buf[pos_in_buf] = *data;
+		pos_in_buf++;
+		data++;
+		length--;
+
+		zassert_not_equal(pos_in_buf, sizeof(test_output_buf)-1,
+				  "Increase the size of test_output_buf");
+	}
+	return output_length;
+}
+
+LOG_OUTPUT_DEFINE(log_output_mock, char_out, mock_output_buf, sizeof(mock_output_buf));
+
+static void process(const struct log_backend *const backend,
+		union log_msg2_generic *msg)
+{
+	uint32_t flags = log_backend_std_get_flags();
+
+	test_msg = msg;
+
+	log_format_func_t log_output_func = log_format_func_t_get(log_format_current);
+
+	log_output_func(&log_output_mock, &msg->log, flags);
+}
+
+bool isKthBitSet(int n, int k)
+{
+	return (n & (1 << (k - 1)));
```

Comment:
Very minor quibble. If returning a bool, this should be using something like ...

    return ((n & (1 << (k - 1))) != 0);

... otherwise it is returning an integer.

Reply:
Good catch. Thanks. Addressed!!